terraform/transform_provider.go

```diff
@@ -195,8 +195,8 @@ func (t *MissingProviderTransformer) Transform(g *Graph) error {
 				continue
 			}
 
-			// Add our own missing provider node to the graph
-			m[p] = g.Add(&graphNodeMissingProvider{ProviderNameValue: p})
+			// Add the missing provider node to the graph
+			m[p] = g.Add(&graphNodeProvider{ProviderNameValue: p})
```

Comment:
Yeah, that was exactly the argument for renaming this... Since this node represents the provider itself in the graph, this name seemed a little more suitable and more descriptive.

I added the rename in this PR as the provisioner and provider transformers are very much inline with each other and I changed it for the provisioner as well. 

But I'll take this code out if you prefer that and make a new PR for the provider renaming... :+1: